dataset/include/scipp/dataset/dataset.h

```diff
@@ -684,6 +703,8 @@ class SCIPP_DATASET_EXPORT DataArray {
 
   void drop_alignment();
 
+  scipp::index sizeInMemory() const { return m_holder.sizeInMemory(); }
```

Comment:
See above. No need to add methods everywhere, I think? `scipp::index sizeInMemory(const DataArrayConstView &)` should cover both `DataArray` and its views through implicit conversion.